src/markov.rs

```diff
@@ -1,9 +1,10 @@
-use rand::{thread_rng, Rng};
 use std::collections::HashMap;
+use rand::{thread_rng, Rng};
 use std::hash::Hash;
 use typemap::Key;
+use regex::Regex;
 
-fn roulette_wheel<'a, T: Eq + Hash>(map: &'a HashMap<T, u32>, rng: &mut Rng) -> Option<&'a T> {
+fn get_next_word<'a, T: Eq + Hash>(map: &'a HashMap<T, u32>, rng: &mut Rng) -> Option<&'a T> {
```

Comment:
:-1: 

This is not specific to getting words, but rather choosing _any_ item of type `T` from a weighted `HashMap`